File: University/HalazuneMokhtasatiS1.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {

            int point = scanner.nextInt();

            if ((point + 1) % 4 == 0) {
                int n = (point + 1)/4;
                System.out.print(n + " " + n);
            }else if ((point) % 4 == 0) {
                int n = (point)/4;
                System.out.print((-1) * n + " " + n);
            }else if ((point - 1) % 4 == 0) {
                int n = (point - 1)/4;
                System.out.print((-1) * n + " " + (-1) * n);
            }else {
                int n = (point + 2)/4;
                System.out.print(n + " " + ((-1) * n + 1));
            }
        }
    }
}

// </Ctrl+Z>

File: University/HalazuneMokhtasatiS1.py
if __name__ == "__main__":
    point = int(input())
    if (point + 1) % 4 == 0:
        n = (point + 1) // 4
        print(n, n)
    elif point % 4 == 0:
        n = point // 4
        print(-1 * n, n)
    elif (point - 1) % 4 == 0:
        n = (point - 1) // 4
        print(-1 * n, -1 * n)
    else:
        n = (point + 2) // 4
        print(n, -1 * n + 1)
